File: tests/fixtures/repository-pos/impl.js
// SMELL: the service function reaches straight into a data store, building a
// query inline. Persistence and business logic are entangled. Candidate for
// extracting a Repository.
const DB = {
  users: [
    { id: 1, name: "Ada", active: true },
    { id: 2, name: "Lin", active: false },
  ],
};

function getActiveUserName(id) {
  // inline "query"
  const row = DB.users.filter((u) => u.id === id && u.active === true)[0];
  if (!row) return null;
  return row.name.toUpperCase();
}

module.exports = { getActiveUserName };

File: tests/fixtures/py-repository-pos/impl.py
# SMELL: the service function reaches straight into a data store, building a
# query inline. Persistence and business logic are entangled. Candidate for
# extracting a Repository.

DB = {
    "users": [
        {"id": 1, "name": "Ada", "active": True},
        {"id": 2, "name": "Lin", "active": False},
    ]
}


def get_active_user_name(user_id: int):
    # inline "query" interleaved with business logic
    rows = [u for u in DB["users"] if u["id"] == user_id and u["active"] is True]
    if not rows:
        return None
    return rows[0]["name"].upper()
